### ISSUE TITLE
Streaming tokens?

### ISSUE THREAD

**Issue by LoopControl** [henryhuangh/gpt_chatbot#1]
Just found your project while searching for an API that has autogptq support and you're literally the only project doing so.

Before I try to run your script, can you tell me if this project has support for streaming the model.generate results via the API (instead of waiting for all tokens to finish generating before turning the API response)?

**Comment by henryhuangh** [comment]
Oh really, because [text-generation-webui](https://github.com/oobabooga/text-generation-webui) is doing something similar and has gptq support. This does not currently support streaming but I plan to add it in a couple of days!

**Comment by LoopControl** [comment]
Yep I use text-gen-ui personally and it’s great but I meant you’re the only project that seems to be serving a gptq model via API (at least when I do a GitHub code search). 

Even though text-gen has an API, it can’t handle parallel generations (literally just kills the websocket connection when you start a 2nd generation while first generation is running) and has a lot of overhead for simple API-based/programmatic inference

**Comment by henryhuangh** [comment]
Yes, I totally understand it is frustrating to not be able to find just a simple API implementation without the unnecessary layers! I wanted to create this for that reason. May I ask what your use case is, you might find this useful [simpleAI](https://github.com/lhenault/simpleAI).

I updated the text_generation endpoint to support data streaming. Still figuring out how to proxy the streamed responses using langchain.
